File: packages/flutter/lib/src/resources/health.dart
/// Health resource — `/health` (outside /api).
library;

import '../models.dart';
import 'base.dart';

/// Provides access to the `/health` endpoint.
class HealthResource extends BaseResource {
  const HealthResource(super.client);

  /// `GET /health` — returns the server health status.
  Future<HealthStatus> get() async {
    final data = await client.request('GET', 'health', api: false);
    return HealthStatus.fromJson((data as Map<String, dynamic>?) ?? {});
  }
}

File: packages/python/src/tbd_agents/resources/health.py
from __future__ import annotations

from tbd_agents.types import HealthStatus

from .base import BaseResource


class HealthResource(BaseResource):
    def get(self) -> HealthStatus:
        data = self._client.request("GET", "/health", api=False)
        return self._parse_model(data, HealthStatus)

